client/src/app/components/home/home.component.ts

```diff
@@ -1,20 +1,45 @@
-import { Component, OnInit } from '@angular/core';
+import { MediaMatcher } from '@angular/cdk/layout';
+import { AfterViewInit, ChangeDetectorRef, Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { MatSidenav } from '@angular/material/sidenav';
 import { TokenService } from 'src/app/services/token.service';
 import { Router } from '@angular/router';
+import { SidenavService } from 'src/app/services/sidenav.service';
 
 @Component({
   selector: 'app-home',
   templateUrl: './home.component.html',
   styleUrls: ['./home.component.css'],
 })
-export class HomeComponent implements OnInit {
-  constructor(private tokenService: TokenService, private router: Router) {}
+export class HomeComponent implements OnDestroy, AfterViewInit, OnInit {
+  mobileQuery: MediaQueryList;
+  private mobileQueryListener: () => void;
+  @ViewChild('sidenav', { static: true }) sidenav: MatSidenav;
 
-  ngOnInit(): void {}
+  constructor(
+    changeDetectorRef: ChangeDetectorRef,
+    media: MediaMatcher,
+    private tokenService: TokenService,
+    private router: Router,
+    private sidenavService: SidenavService
+  ) {
+    this.mobileQuery = media.matchMedia('(max-width: 600px)');
+    this.mobileQueryListener = () => changeDetectorRef.detectChanges();
+    this.mobileQuery.addEventListener('change', this.mobileQueryListener);
+  }
+
+  ngOnInit(): void { }
+
+  ngOnDestroy(): void {
+    this.mobileQuery.removeEventListener('change', this.mobileQueryListener);
+  }
+
+  ngAfterViewInit(): void {
+    this.sidenavService.setSidenav(this.sidenav);
+  }
 
   logout(): void {
```

Comment:
Can you please delete this method since now it's implemented inside toolbar component.

Reply:
This is actually needed here as the sidenav element is inside the home component.

Reply:
No i'm talking about the logout().

Reply:
yes sorry